### PR TITLE
Add bootstrap test with -Dcassandra.write_survey and nodetool join

### PR DESCRIPTION
**Pull request by pauloricardomg**
Test for CASSANDRA-11379

### CODE REVIEW

**Comment by pauloricardomg** [comment]
just closed CASSANDRA-11379 as "Not a Problem" but this test is still relevant to validate write_survey mode and nodetool join

File: bootstrap_test.py [hunk 1]
```diff
@@ -90,6 +93,30 @@ def simple_bootstrap_test(self):
...
+
+    def bootstrap_on_write_survey_test(self):
+        def bootstrap_on_write_survey_and_join(cluster, token):
+            node2 = new_node(cluster)
+            node2.set_configuration_options(values={'initial_token': token})
+            node2.start(jvm_args=["-Dcassandra.write_survey=true"], wait_for_binary_proto=True)
+            self.assertTrue(len(node2.grep_log('Startup complete, but write survey mode is active, not becoming an active ring member.')) > 0)
+            self.check_bootstrap_state(node2, 'IN_PROGRESS')
+            node2.nodetool("join")
+            self.assertTrue(len(node2.grep_log('Leaving write survey mode and joining ring at operator request')) > 0)
```
Comment (ptnapoleon) [inline bootstrap_test.py]: Can you use `self.assertGreater` here instead? It provides better debug output. Same with line 112

**Comment by ptnapoleon** [comment]
Can you annotate the new test with `@jira_ticket CASSANDRA-11379` in the docstring?

File: bootstrap_test.py [hunk 2]
```diff
@@ -90,6 +93,30 @@ def simple_bootstrap_test(self):
         assert_almost_equal(size1, size2, error=0.3)
         assert_almost_equal(float(initial_size - empty_size), 2 * (size1 - float(empty_size)))
 
+        self.check_bootstrap_state(node2, 'COMPLETED')
+
+    def simple_bootstrap_test(self):
+        def bootstrap(cluster, token):
+            node2 = new_node(cluster)
+            node2.set_configuration_options(values={'initial_token': token})
```
Comment (ptnapoleon) [inline bootstrap_test.py]: @mambocab It seems this test always (and has always) runs without vnodes. Should we just annotate it as such, to not mislead anyone?

Comment (mambocab) [inline bootstrap_test.py]: +1

**Comment by pauloricardomg** [comment]
Sorry for the lack of update here, but this is still relevant. I'm waiting to rebase this on top of #1051 and address the comments before resubmitting.

**Comment by pauloricardomg** [comment]
Replaced by #1064.

**Comment by ptnapoleon** [comment]
Pinged 1-1, we're still doing this PR, just as #1064

**Comment by ptnapoleon** [comment]
Pinged 1-1, we're still doing this PR, just as #1064

**Pull request closed without merging**